tests/component/test_stream_writers_do.py

```diff
@@ -0,0 +1,615 @@
+    task.do_channels.add_do_chan(
+        real_x_series_device.do_lines[0].name, line_grouping=LineGrouping.CHAN_FOR_ALL_LINES
+    )
+    _start_do_task(task)
+    return task
+
+
+@pytest.fixture
+def do_single_channel_multi_line_task(
+    generate_task: Callable[[], nidaqmx.Task], real_x_series_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    task = generate_task()
+    chan = task.do_channels.add_do_chan(
+        flatten_channel_string(real_x_series_device.do_lines[:8].name),
+        line_grouping=LineGrouping.CHAN_FOR_ALL_LINES,
+    )
+    _start_do_task(task, num_chans=chan.do_num_lines)
+    return task
+
+
+@pytest.fixture
+def do_multi_channel_multi_line_task(
+    generate_task: Callable[[], nidaqmx.Task], real_x_series_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    task = generate_task()
+    task.do_channels.add_do_chan(
+        flatten_channel_string(real_x_series_device.do_lines[:8].name),
+        line_grouping=LineGrouping.CHAN_PER_LINE,
+    )
+    _start_do_task(task, num_chans=task.number_of_channels)
+    return task
+
+
+@pytest.fixture
+def do_port0_task(
+    generate_task: Callable[[], nidaqmx.Task], real_x_series_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    task = generate_task()
+    # X Series port 0 has either 32 or 8 lines. The former can only be used with 32-bit writes. The
+    # latter can be used with any sized port write.
+    task.do_channels.add_do_chan(
+        real_x_series_device.do_ports[0].name,
+        line_grouping=LineGrouping.CHAN_FOR_ALL_LINES,
+    )
+    _start_do_task(task, is_port=True)
+    return task
+
+
+@pytest.fixture
+def do_port1_task(
+    generate_task: Callable[[], nidaqmx.Task], real_x_series_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    task = generate_task()
+    # X Series port 1 has 8 lines, and can be used with any sized port write.
+    task.do_channels.add_do_chan(
+        real_x_series_device.do_ports[1].name,
+        line_grouping=LineGrouping.CHAN_FOR_ALL_LINES,
+    )
+    _start_do_task(task, is_port=True)
+    return task
+
+
+@pytest.fixture
+def do_multi_channel_port_task(
+    generate_task: Callable[[], nidaqmx.Task], real_x_series_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    task = generate_task()
+    # X Series port 1 has 8 lines, and can be used with any sized port write
+    task.do_channels.add_do_chan(
+        real_x_series_device.do_ports[1].name,
+        line_grouping=LineGrouping.CHAN_FOR_ALL_LINES,
+    )
+    # X Series port 2 has 8 lines, and can be used with any sized port write
+    task.do_channels.add_do_chan(
+        real_x_series_device.do_ports[2].name,
+        line_grouping=LineGrouping.CHAN_FOR_ALL_LINES,
+    )
+    _start_do_task(task, is_port=True, num_chans=task.number_of_channels)
+    return task
+
+
+def _start_di_task(task: nidaqmx.Task) -> None:
+    # Don't reserve the lines, so we can read what DO is writing.
+    for chan in task.di_channels:
+        chan.di_tristate = False
```

Comment:
You don't have to iterate through channels. Use `.all` to access all channels together.
```suggestion
    task.di_channels.all.di_tristate = False
```